akka-projection-cassandra/src/test/scala/akka/projection/cassandra/CassandraProjectionSpec.scala

```diff
@@ -494,6 +515,67 @@ class CassandraProjectionSpec
         handler.attempts shouldBe 1 + 3
       }
     }
+
+    "verify offsets before processing an envelope" in {
+      val entityId = UUID.randomUUID().toString
+      val projectionId = genRandomProjectionId()
+      val (verifiedQueue, verifiedProbe) = Source
+        .queue[Long](1, OverflowStrategy.backpressure)
+        .toMat(TestSink.probe(system.classicSystem))(Keep.both)
+        .run()
+
+      val testVerification = (offset: Long) => {
+        Await.ready(verifiedQueue.offer(offset), 10.millis)
```

Comment:
I wanted to add artificial blocking here to control the rate of the stream.  The streams sink probe doesn't give me enough control, because various async  operators in the stream can buffer messages and make it difficult to recreate the flow that I need.  Using queues let's me control the rate of flow (if I block on `offer`).

I can probably do the same with a probe, but I haven't done it before so I reached for `Source.queue`.

Reply:
I removed some usages of the queue where discussed.